sway-lsp/src/capabilities/rename.rs

```diff
@@ -148,29 +145,32 @@ fn is_token_in_workspace(
     engines: &Engines,
     token: &Token,
 ) -> Result<bool, LanguageServerError> {
-    let decl_span = token
-        .declared_token_span(engines)
+    let decl_ident = token
+        .declared_token_ident(engines)
         .ok_or(RenameError::TokenNotFound)?;
 
     // Check the span of the tokens defintions to determine if it's in the users workspace.
     let temp_path = &session.sync.temp_dir()?;
-    if let Some(id) = decl_span.source_id() {
-        let path = engines.se().get_path(id);
+    if let Some(path) = decl_ident.path {
         if !path.starts_with(temp_path) {
             return Err(LanguageServerError::RenameError(
                 RenameError::TokenNotPartOfWorkspace,
             ));
         }
     }
```

Comment:
nit: It would be nice to move this into a helper function in `sync.rs` with a comment. It took me a minute to understand the difference between what this is checking and what `sync.is_path_in_temp_workspace()` does. I believe this is checking that it's in a session subdirectory within the temp directory, right?

Does `TokenIdent.path` always prefer the temp path if it exists?

Reply:
I guess the idea behind this function is to make sure that we don't allow renaming any tokens that are defined outside of the users project. So types from external libraries don't get renamed. The path in TokenIdent always points to temp memory so that it's in sync with the clients edits. 

This change here just removes the need to call into the source engine to retrieve the path since we have already extracted it in the traversal stage.